gateclient/gateclient.go

```diff
@@ -50,6 +50,9 @@ func NewGateapiClient(floodgateConfig *config.Config) *GateapiClient {
 
 // WaitForSuccessfulTask function is waiting for task to finish
 func (c GateapiClient) WaitForSuccessfulTask(checkTask map[string]interface{}, maxRetries int) error {
+	if checkTask == nil {
+		return fmt.Errorf("task is empty, check if Spinnaker credentials are valid")
```

Comment:
If there are more instances when this list can be empty the message should be more generic.

Reply:
I want to keep the Spinnaker credentials annotation. Will rephrasing it as "task to check is empty. this can be caused by invalid Spinnaker credentials" suffice?

Reply:
@mwierzbicki Are you able to add in error which spinnaker address was used? It should be enough to locate which spinnaker instance we were trying to connect to.

Reply:
I think we should have some way to check if the credentials are valid before we start applying changes.
Another thing: credentials can be valid but the user can just not have enough permissions if RBAC is configured. I just don't think this is a good way of handling that issue.

Reply:
https://www.spinnaker.io/reference/api/docs.html#api-Authcontroller-userUsingGET maybe this endpoint can be used to check credentials first?

Reply:
^ nice find. I'll look into that.

Reply:
Done. I used https://www.spinnaker.io/reference/api/docs.html#api-Authcontroller-loggedOutUsingGET.